client/src/js/components/container/Signup.jsx

```diff
@@ -0,0 +1,210 @@
+import React, { Component } from 'react';
+import toastr from 'toastr';
+
+import AppActions from '../../actions/AppActions';
+import AppStore from '../../stores/AppStore';
+import { validateEmail } from './../../helpers/utils';
+import Input from '../presentation/Input';
+
+
+/**
+ * @description the signup components lets users register an account
+ *
+ * @class SignUp
+ *
+ * @extends {Component}
+ */
+export default class Signup extends Component {
+ /**
+   * @description Creates an instance of Signup.
+   * bind methods and set initial state.
+	 *
+   * @memberof Signup
+   *
+   * @param {object} props
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      contacts: [],
+      databaseUsers: [],
+      emails: [],
+      numbers: [],
+      googleUser: null,
+      username: '',
+      email: '',
+      number: '',
+      password: '',
+      verifyPassword: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+    /**
+     * @method componentDidMount
+     *
+     * @description Adds an event Listener to the
+	   * Store and fires when the component is fully mounted.
+     *
+     * @return {void} void
+     *
+     * @memberof Signup
+     */
+  componentDidMount() {
+    AppStore.addChangeListener(this.onChange);
+  }
+
+    /**
+    * @method componentWillUnmount
+    *
+		* @description Removes event Listener from the Store
+		*
+		* @return {void} void
+    *
+    * @memberof Signup
+    */
+  componentWillUnmount() {
+    AppStore.removeChangeListener(this.onChange);
+  }
+
+    /**
+     * @description: controls inputs state
+    *
+    * @param {Object} element the current element
+    *
+    * @return {void} void
+    */
+  handleChange(element) {
+    this.setState({
+      [element.target.name]: element.target.value
+    });
+  }
+
+    /**
+     * @method onChange
+     *
+     * @description this method gets data from the store and sets to state
+		 *
```

Comment:
The indentation for your comments looks off, kindly fix it and it occurs in several files too

Reply:
I have fixed this